src/game.cpp

```diff
@@ -12592,6 +12592,10 @@ void game::plswim(int x, int y)
         add_msg(_("The water puts out the flames!"));
         u.remove_effect("onfire");
     }
+    if (u.has_effect("onfire")) {
```

Comment:
That glass ball is 100% correct! It must be magic ;)
And I'll look at `player::active_light`. That sounds like exactly what I was looking for. Would it still work on NPCs though?